gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -316,12 +341,14 @@ public boolean equals(Object o) {
         && Objects.equals(callOptions, that.callOptions)
         && Objects.equals(streamWaitTimeout, that.streamWaitTimeout)
         && Objects.equals(streamIdleTimeout, that.streamIdleTimeout)
-        && Objects.equals(channelAffinity, that.channelAffinity);
+        && Objects.equals(channelAffinity, that.channelAffinity)
+        && Objects.equals(extraHeaders, that.extraHeaders);
```

Comment:
I just realized `Metadata` doesn't override `equals`. I don't think this will work, since two "identical" `Metadata`s will make us compare unequal. I wrote a small doc about this and tagged you.